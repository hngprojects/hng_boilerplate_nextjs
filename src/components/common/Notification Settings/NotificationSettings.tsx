"use client";

import { Check } from "lucide-react";
import { useEffect, useState } from "react";

import CustomButton from "~/components/common/Button/button";
import NotificationSavedModal from "~/components/modals/NotificationSettingSavedModal";
import { Switch } from "~/components/ui/switch";

const LOCAL_STORAGE_KEY = "notificationSettings";

interface Settings {
  pushNotifications: boolean;
  emailActivity: boolean;
  emailAlways: boolean;
  emailDigests: boolean;
  emailAnnouncements: boolean;
  slackActivity: boolean;
  slackAlways: boolean;
  slackAnnouncements: boolean;
}

interface SettingItem {
  label: string;
  description: string;
  key: keyof Settings;
}

const SETTINGS_DATA: { category: string; settings: SettingItem[] }[] = [
  {
    category: "Notifications Alert",
    settings: [
      {
        label: "Mobile push notifications",
        description:
          "Receive push notifications on mentions and comments via your mobile app",
        key: "pushNotifications",
      },
    ],
  },
  {
    category: "Email Notifications",
    settings: [
      {
        label: "Activity in your workspace",
        description:
          "Receive emails when you get comments, mentions, page invites, reminders, access requests, and property changes",
        key: "emailActivity",
      },
      {
        label: "Always send email notifications",
        description:
          "Receive emails about activity in your workspace, even when you are active on the app",
        key: "emailAlways",
      },
      {
        label: "Email digests",
        description:
          "Receive email digest every 8 hours for changes to pages you are subscribed to",
        key: "emailDigests",
      },
      {
        label: "Announcements and update emails",
        description:
          "Receive occasional emails about product launches and new features from Notion",
        key: "emailAnnouncements",
      },
    ],
  },
  {
    category: "Slack notifications",
    settings: [
      {
        label: "Activity in your workspace",
        description:
          "Receive emails when you get comments, mentions, page invites, reminders, access requests, and property changes",
        key: "slackActivity",
      },
      {
        label: "Always send email notifications",
        description:
          "Receive emails about activity in your workspace, even when you are active on the app",
        key: "slackAlways",
      },
      {
        label: "Announcements and update emails",
        description:
          "Receive occasional emails about product launches and new features from Notion",
        key: "slackAnnouncements",
      },
    ],
  },
];

export default function NotificationSettings() {
  const [settings, setSettings] = useState<Settings>({
    pushNotifications: false,
    emailActivity: false,
    emailAlways: false,
    emailDigests: false,
    emailAnnouncements: false,
    slackActivity: false,
    slackAlways: false,
    slackAnnouncements: false,
  });

  const [temporarySettings, setTemporarySettings] =
    useState<Settings>(settings);

  useEffect(() => {
    const savedSettings = localStorage.getItem(LOCAL_STORAGE_KEY);
    if (savedSettings) {
      const parsedSettings = JSON.parse(savedSettings);
      setSettings(parsedSettings);
      setTemporarySettings(parsedSettings);
    }
  }, []);

  const [isModalOpen, setIsModalOpen] = useState(false);

  const handleToggle = (setting: keyof Settings) => {
    setTemporarySettings((previousSettings) => ({
      ...previousSettings,
      [setting]: !previousSettings[setting],
    }));
  };

  const handleSaveChanges = () => {
    setSettings(temporarySettings);
    localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify(temporarySettings));
    setIsModalOpen(true);
  };

  const closeModal = () => setIsModalOpen(false);

  return (
    <div className="flex w-full bg-default-foreground">
      <div className="w-full py-8 pl-8 pr-28">
        {SETTINGS_DATA.map((section) => (
          <div key={section.category} className="mb-6">
            <h3 className="mb-4 text-base font-bold sm:text-lg lg:text-2xl">
              {section.category}
            </h3>
            {section.settings.map((setting) => (
              <div key={setting.key} className="mb-3">
                <h4
                  data-testid={`switch-${setting.key}`}
                  className="text-sm font-medium"
                >
                  {setting.label}
                </h4>
                <div className="flex flex-col justify-between gap-2 text-sm sm:flex-row sm:text-base lg:text-lg">
                  <p className="text-sm text-gray-600">{setting.description}</p>
                  <div
                    onClick={() => handleToggle(setting.key)}
                    data-testid="switch-pushNotifications"
                  >
                    <Switch
                      role="switch"
                      data-testid={`switch-${setting.key}`}
                      checked={temporarySettings[setting.key]}
                      onCheckedChange={() => handleToggle(setting.key)}
                    />
                  </div>
                </div>
              </div>
            ))}
          </div>
        ))}
        <div
          onClick={handleSaveChanges}
          className="mt-6 flex w-full justify-end"
        >
          <CustomButton variant="primary">
            <Check /> Save Changes
          </CustomButton>
        </div>
      </div>
      <NotificationSavedModal show={isModalOpen} onClose={closeModal} />
    </div>
  );
}
