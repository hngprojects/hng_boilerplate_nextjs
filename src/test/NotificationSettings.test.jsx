import { fireEvent, render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import NotificationSettings from "../app/settings/notification-settings/notificationsettings";

const SETTINGS_DATA = [
  {
    category: "Notifications Alert",
    settings: [
      {
        label: "Mobile push notifications",
        description:
          "Receive push notifications on mentions and comments via your mobile app",
        key: "pushNotifications",
      },
    ],
  },
  {
    category: "Email Notifications",
    settings: [
      {
        label: "Activity in your workspace",
        description:
          "Receive emails when you get comments, mentions, page invites, reminders, access requests, and property changes",
        key: "emailActivity",
      },
      {
        label: "Always send email notifications",
        description:
          "Receive emails about activity in your workspace, even when you are active on the app",
        key: "emailAlways",
      },
      {
        label: "Email digests",
        description:
          "Receive email digest every 8 hours for changes to pages you are subscribed to",
        key: "emailDigests",
      },
      {
        label: "Announcements and update emails",
        description:
          "Receive occasional emails about product launches and new features from Notion",
        key: "emailAnnouncements",
      },
    ],
  },
  {
    category: "Slack notifications",
    settings: [
      {
        label: "Activity in your workspace",
        description:
          "Receive emails when you get comments, mentions, page invites, reminders, access requests, and property changes",
        key: "slackActivity",
      },
      {
        label: "Always send email notifications",
        description:
          "Receive emails about activity in your workspace, even when you are active on the app",
        key: "slackAlways",
      },
      {
        label: "Announcements and update emails",
        description:
          "Receive occasional emails about product launches and new features from Notion",
        key: "slackAnnouncements",
      },
    ],
  },
];

describe("notificationSettings Component", () => {
  it("renders all sections and settings", () => {
    expect.assertions(19);
    render(<NotificationSettings />);
    for (const section of SETTINGS_DATA) {
      expect(screen.getByText(section.category)).toBeInTheDocument();
      for (const setting of section.settings) {
        const labels = screen.getAllByText(setting.label);
        expect(labels.length).toBeGreaterThan(0);
        const descriptions = screen.getAllByText(setting.description);
        expect(descriptions.length).toBeGreaterThan(0);
      }
    }
  });

  it("saves changes and displays a confirmation message", () => {
    expect.assertions(1);
    render(<NotificationSettings />);
    const saveButton = screen.getByText(/save changes/i);
    fireEvent.click(saveButton);
    expect(screen.queryByText(/settings saved!/i)).not.toBeInTheDocument();
  });

  it("closes the modal when Done button is clicked", () => {
    expect.assertions(1);
    render(<NotificationSettings />);
    fireEvent.click(screen.getByText(/save changes/i));
    fireEvent.click(screen.getByText(/done/i));
    expect(screen.queryByText(/settings saved!/i)).not.toBeInTheDocument();
  });
});
