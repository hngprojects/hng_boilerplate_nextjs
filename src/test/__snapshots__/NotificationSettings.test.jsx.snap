// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home Component > renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="p-4 sm:p-6 lg:p-8"
  >
    <div
      class="mb-6"
    >
      <h3
        class="text-base font-semibold sm:text-lg lg:text-xl"
      >
        Notifications Alert
      </h3>
      <div
        class="mb-4"
      >
        <h4
          class="text-sm font-medium sm:text-base lg:text-lg"
        >
          Mobile push notifications
        </h4>
        <div
          class="flex flex-col justify-between gap-2 text-sm sm:flex-row sm:text-base lg:text-lg"
        >
          <p
            class="text-gray-600"
          >
            Receive push notifications on mentions and comments via your mobile app
          </p>
          <div>
            <button
              aria-checked="false"
              class="peer inline-flex h-6 w-11 shrink-0 cursor-pointer items-center rounded-full border-2 border-transparent transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 focus-visible:ring-offset-background disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=unchecked]:bg-input"
              data-state="unchecked"
              role="switch"
              type="button"
              value="on"
            >
              <span
                class="pointer-events-none block h-5 w-5 rounded-full bg-background shadow-lg ring-0 transition-transform data-[state=checked]:translate-x-5 data-[state=unchecked]:translate-x-0"
                data-state="unchecked"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mb-6"
    >
      <h3
        class="text-base font-semibold sm:text-lg lg:text-xl"
      >
        Email Notifications
      </h3>
      <div
        class="mb-4"
      >
        <h4
          class="text-sm font-medium sm:text-base lg:text-lg"
        >
          Activity in your workspace
        </h4>
        <div
          class="flex flex-col justify-between gap-2 text-sm sm:flex-row sm:text-base lg:text-lg"
        >
          <p
            class="text-gray-600"
          >
            Receive emails when you get comments, mentions, page invites, reminders, access requests, and property changes
          </p>
          <div>
            <button
              aria-checked="false"
              class="peer inline-flex h-6 w-11 shrink-0 cursor-pointer items-center rounded-full border-2 border-transparent transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 focus-visible:ring-offset-background disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=unchecked]:bg-input"
              data-state="unchecked"
              role="switch"
              type="button"
              value="on"
            >
              <span
                class="pointer-events-none block h-5 w-5 rounded-full bg-background shadow-lg ring-0 transition-transform data-[state=checked]:translate-x-5 data-[state=unchecked]:translate-x-0"
                data-state="unchecked"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="mb-4"
      >
        <h4
          class="text-sm font-medium sm:text-base lg:text-lg"
        >
          Always send email notifications
        </h4>
        <div
          class="flex flex-col justify-between gap-2 text-sm sm:flex-row sm:text-base lg:text-lg"
        >
          <p
            class="text-gray-600"
          >
            Receive emails about activity in your workspace, even when you are active on the app
          </p>
          <div>
            <button
              aria-checked="false"
              class="peer inline-flex h-6 w-11 shrink-0 cursor-pointer items-center rounded-full border-2 border-transparent transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 focus-visible:ring-offset-background disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=unchecked]:bg-input"
              data-state="unchecked"
              role="switch"
              type="button"
              value="on"
            >
              <span
                class="pointer-events-none block h-5 w-5 rounded-full bg-background shadow-lg ring-0 transition-transform data-[state=checked]:translate-x-5 data-[state=unchecked]:translate-x-0"
                data-state="unchecked"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="mb-4"
      >
        <h4
          class="text-sm font-medium sm:text-base lg:text-lg"
        >
          Email digests
        </h4>
        <div
          class="flex flex-col justify-between gap-2 text-sm sm:flex-row sm:text-base lg:text-lg"
        >
          <p
            class="text-gray-600"
          >
            Receive email digest every 8 hours for changes to pages you are subscribed to
          </p>
          <div>
            <button
              aria-checked="false"
              class="peer inline-flex h-6 w-11 shrink-0 cursor-pointer items-center rounded-full border-2 border-transparent transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 focus-visible:ring-offset-background disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=unchecked]:bg-input"
              data-state="unchecked"
              role="switch"
              type="button"
              value="on"
            >
              <span
                class="pointer-events-none block h-5 w-5 rounded-full bg-background shadow-lg ring-0 transition-transform data-[state=checked]:translate-x-5 data-[state=unchecked]:translate-x-0"
                data-state="unchecked"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="mb-4"
      >
        <h4
          class="text-sm font-medium sm:text-base lg:text-lg"
        >
          Announcements and update emails
        </h4>
        <div
          class="flex flex-col justify-between gap-2 text-sm sm:flex-row sm:text-base lg:text-lg"
        >
          <p
            class="text-gray-600"
          >
            Receive occasional emails about product launches and new features from Notion
          </p>
          <div>
            <button
              aria-checked="false"
              class="peer inline-flex h-6 w-11 shrink-0 cursor-pointer items-center rounded-full border-2 border-transparent transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 focus-visible:ring-offset-background disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=unchecked]:bg-input"
              data-state="unchecked"
              role="switch"
              type="button"
              value="on"
            >
              <span
                class="pointer-events-none block h-5 w-5 rounded-full bg-background shadow-lg ring-0 transition-transform data-[state=checked]:translate-x-5 data-[state=unchecked]:translate-x-0"
                data-state="unchecked"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mb-6"
    >
      <h3
        class="text-base font-semibold sm:text-lg lg:text-xl"
      >
        Slack notifications
      </h3>
      <div
        class="mb-4"
      >
        <h4
          class="text-sm font-medium sm:text-base lg:text-lg"
        >
          Activity in your workspace
        </h4>
        <div
          class="flex flex-col justify-between gap-2 text-sm sm:flex-row sm:text-base lg:text-lg"
        >
          <p
            class="text-gray-600"
          >
            Receive emails when you get comments, mentions, page invites, reminders, access requests, and property changes
          </p>
          <div>
            <button
              aria-checked="false"
              class="peer inline-flex h-6 w-11 shrink-0 cursor-pointer items-center rounded-full border-2 border-transparent transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 focus-visible:ring-offset-background disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=unchecked]:bg-input"
              data-state="unchecked"
              role="switch"
              type="button"
              value="on"
            >
              <span
                class="pointer-events-none block h-5 w-5 rounded-full bg-background shadow-lg ring-0 transition-transform data-[state=checked]:translate-x-5 data-[state=unchecked]:translate-x-0"
                data-state="unchecked"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="mb-4"
      >
        <h4
          class="text-sm font-medium sm:text-base lg:text-lg"
        >
          Always send email notifications
        </h4>
        <div
          class="flex flex-col justify-between gap-2 text-sm sm:flex-row sm:text-base lg:text-lg"
        >
          <p
            class="text-gray-600"
          >
            Receive emails about activity in your workspace, even when you are active on the app
          </p>
          <div>
            <button
              aria-checked="false"
              class="peer inline-flex h-6 w-11 shrink-0 cursor-pointer items-center rounded-full border-2 border-transparent transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 focus-visible:ring-offset-background disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=unchecked]:bg-input"
              data-state="unchecked"
              role="switch"
              type="button"
              value="on"
            >
              <span
                class="pointer-events-none block h-5 w-5 rounded-full bg-background shadow-lg ring-0 transition-transform data-[state=checked]:translate-x-5 data-[state=unchecked]:translate-x-0"
                data-state="unchecked"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="mb-4"
      >
        <h4
          class="text-sm font-medium sm:text-base lg:text-lg"
        >
          Announcements and update emails
        </h4>
        <div
          class="flex flex-col justify-between gap-2 text-sm sm:flex-row sm:text-base lg:text-lg"
        >
          <p
            class="text-gray-600"
          >
            Receive occasional emails about product launches and new features from Notion
          </p>
          <div>
            <button
              aria-checked="false"
              class="peer inline-flex h-6 w-11 shrink-0 cursor-pointer items-center rounded-full border-2 border-transparent transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 focus-visible:ring-offset-background disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=unchecked]:bg-input"
              data-state="unchecked"
              role="switch"
              type="button"
              value="on"
            >
              <span
                class="pointer-events-none block h-5 w-5 rounded-full bg-background shadow-lg ring-0 transition-transform data-[state=checked]:translate-x-5 data-[state=unchecked]:translate-x-0"
                data-state="unchecked"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mt-6"
    >
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground shadow h-9 px-4 py-2"
        role="button"
      >
        <svg
          class="lucide lucide-check"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 6 9 17l-5-5"
          />
        </svg>
         Save Changes
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`NotificationSettings Component > renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="p-4 sm:p-6 lg:p-8"
  >
    <div
      class="mb-6"
    >
      <h3
        class="text-base font-semibold sm:text-lg lg:text-xl"
      >
        Notifications Alert
      </h3>
      <div
        class="mb-4"
      >
        <h4
          class="text-sm font-medium sm:text-base lg:text-lg"
        >
          Mobile push notifications
        </h4>
        <div
          class="flex flex-col justify-between gap-2 text-sm sm:flex-row sm:text-base lg:text-lg"
        >
          <p
            class="text-gray-600"
          >
            Receive push notifications on mentions and comments via your mobile app
          </p>
          <div>
            <button
              aria-checked="false"
              class="peer inline-flex h-6 w-11 shrink-0 cursor-pointer items-center rounded-full border-2 border-transparent transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 focus-visible:ring-offset-background disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=unchecked]:bg-input"
              data-state="unchecked"
              role="switch"
              type="button"
              value="on"
            >
              <span
                class="pointer-events-none block h-5 w-5 rounded-full bg-background shadow-lg ring-0 transition-transform data-[state=checked]:translate-x-5 data-[state=unchecked]:translate-x-0"
                data-state="unchecked"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mb-6"
    >
      <h3
        class="text-base font-semibold sm:text-lg lg:text-xl"
      >
        Email Notifications
      </h3>
      <div
        class="mb-4"
      >
        <h4
          class="text-sm font-medium sm:text-base lg:text-lg"
        >
          Activity in your workspace
        </h4>
        <div
          class="flex flex-col justify-between gap-2 text-sm sm:flex-row sm:text-base lg:text-lg"
        >
          <p
            class="text-gray-600"
          >
            Receive emails when you get comments, mentions, page invites, reminders, access requests, and property changes
          </p>
          <div>
            <button
              aria-checked="false"
              class="peer inline-flex h-6 w-11 shrink-0 cursor-pointer items-center rounded-full border-2 border-transparent transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 focus-visible:ring-offset-background disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=unchecked]:bg-input"
              data-state="unchecked"
              role="switch"
              type="button"
              value="on"
            >
              <span
                class="pointer-events-none block h-5 w-5 rounded-full bg-background shadow-lg ring-0 transition-transform data-[state=checked]:translate-x-5 data-[state=unchecked]:translate-x-0"
                data-state="unchecked"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="mb-4"
      >
        <h4
          class="text-sm font-medium sm:text-base lg:text-lg"
        >
          Always send email notifications
        </h4>
        <div
          class="flex flex-col justify-between gap-2 text-sm sm:flex-row sm:text-base lg:text-lg"
        >
          <p
            class="text-gray-600"
          >
            Receive emails about activity in your workspace, even when you are active on the app
          </p>
          <div>
            <button
              aria-checked="false"
              class="peer inline-flex h-6 w-11 shrink-0 cursor-pointer items-center rounded-full border-2 border-transparent transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 focus-visible:ring-offset-background disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=unchecked]:bg-input"
              data-state="unchecked"
              role="switch"
              type="button"
              value="on"
            >
              <span
                class="pointer-events-none block h-5 w-5 rounded-full bg-background shadow-lg ring-0 transition-transform data-[state=checked]:translate-x-5 data-[state=unchecked]:translate-x-0"
                data-state="unchecked"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="mb-4"
      >
        <h4
          class="text-sm font-medium sm:text-base lg:text-lg"
        >
          Email digests
        </h4>
        <div
          class="flex flex-col justify-between gap-2 text-sm sm:flex-row sm:text-base lg:text-lg"
        >
          <p
            class="text-gray-600"
          >
            Receive email digest every 8 hours for changes to pages you are subscribed to
          </p>
          <div>
            <button
              aria-checked="false"
              class="peer inline-flex h-6 w-11 shrink-0 cursor-pointer items-center rounded-full border-2 border-transparent transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 focus-visible:ring-offset-background disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=unchecked]:bg-input"
              data-state="unchecked"
              role="switch"
              type="button"
              value="on"
            >
              <span
                class="pointer-events-none block h-5 w-5 rounded-full bg-background shadow-lg ring-0 transition-transform data-[state=checked]:translate-x-5 data-[state=unchecked]:translate-x-0"
                data-state="unchecked"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="mb-4"
      >
        <h4
          class="text-sm font-medium sm:text-base lg:text-lg"
        >
          Announcements and update emails
        </h4>
        <div
          class="flex flex-col justify-between gap-2 text-sm sm:flex-row sm:text-base lg:text-lg"
        >
          <p
            class="text-gray-600"
          >
            Receive occasional emails about product launches and new features from Notion
          </p>
          <div>
            <button
              aria-checked="false"
              class="peer inline-flex h-6 w-11 shrink-0 cursor-pointer items-center rounded-full border-2 border-transparent transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 focus-visible:ring-offset-background disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=unchecked]:bg-input"
              data-state="unchecked"
              role="switch"
              type="button"
              value="on"
            >
              <span
                class="pointer-events-none block h-5 w-5 rounded-full bg-background shadow-lg ring-0 transition-transform data-[state=checked]:translate-x-5 data-[state=unchecked]:translate-x-0"
                data-state="unchecked"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mb-6"
    >
      <h3
        class="text-base font-semibold sm:text-lg lg:text-xl"
      >
        Slack notifications
      </h3>
      <div
        class="mb-4"
      >
        <h4
          class="text-sm font-medium sm:text-base lg:text-lg"
        >
          Activity in your workspace
        </h4>
        <div
          class="flex flex-col justify-between gap-2 text-sm sm:flex-row sm:text-base lg:text-lg"
        >
          <p
            class="text-gray-600"
          >
            Receive emails when you get comments, mentions, page invites, reminders, access requests, and property changes
          </p>
          <div>
            <button
              aria-checked="false"
              class="peer inline-flex h-6 w-11 shrink-0 cursor-pointer items-center rounded-full border-2 border-transparent transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 focus-visible:ring-offset-background disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=unchecked]:bg-input"
              data-state="unchecked"
              role="switch"
              type="button"
              value="on"
            >
              <span
                class="pointer-events-none block h-5 w-5 rounded-full bg-background shadow-lg ring-0 transition-transform data-[state=checked]:translate-x-5 data-[state=unchecked]:translate-x-0"
                data-state="unchecked"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="mb-4"
      >
        <h4
          class="text-sm font-medium sm:text-base lg:text-lg"
        >
          Always send email notifications
        </h4>
        <div
          class="flex flex-col justify-between gap-2 text-sm sm:flex-row sm:text-base lg:text-lg"
        >
          <p
            class="text-gray-600"
          >
            Receive emails about activity in your workspace, even when you are active on the app
          </p>
          <div>
            <button
              aria-checked="false"
              class="peer inline-flex h-6 w-11 shrink-0 cursor-pointer items-center rounded-full border-2 border-transparent transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 focus-visible:ring-offset-background disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=unchecked]:bg-input"
              data-state="unchecked"
              role="switch"
              type="button"
              value="on"
            >
              <span
                class="pointer-events-none block h-5 w-5 rounded-full bg-background shadow-lg ring-0 transition-transform data-[state=checked]:translate-x-5 data-[state=unchecked]:translate-x-0"
                data-state="unchecked"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="mb-4"
      >
        <h4
          class="text-sm font-medium sm:text-base lg:text-lg"
        >
          Announcements and update emails
        </h4>
        <div
          class="flex flex-col justify-between gap-2 text-sm sm:flex-row sm:text-base lg:text-lg"
        >
          <p
            class="text-gray-600"
          >
            Receive occasional emails about product launches and new features from Notion
          </p>
          <div>
            <button
              aria-checked="false"
              class="peer inline-flex h-6 w-11 shrink-0 cursor-pointer items-center rounded-full border-2 border-transparent transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 focus-visible:ring-offset-background disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=unchecked]:bg-input"
              data-state="unchecked"
              role="switch"
              type="button"
              value="on"
            >
              <span
                class="pointer-events-none block h-5 w-5 rounded-full bg-background shadow-lg ring-0 transition-transform data-[state=checked]:translate-x-5 data-[state=unchecked]:translate-x-0"
                data-state="unchecked"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mt-6"
    >
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground shadow h-9 px-4 py-2"
        role="button"
      >
        <svg
          class="lucide lucide-check"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 6 9 17l-5-5"
          />
        </svg>
         Save Changes
      </button>
    </div>
  </div>
</DocumentFragment>
`;
